core/math/octree.h

```diff
@@ -294,14 +294,16 @@ class Octree {
 	void _unpair_element(Element *p_element, Octant *p_octant);
 
 	struct _CullConvexData {
-		const Plane *planes;
-		int plane_count;
-		const Vector3 *points;
-		int point_count;
+		const Vector<Plane> planes;
+		const Vector<Vector3> &points;
```

Comment:
Isn't it still dangerous to store a reference to a Vector for `points`? It seems it's allocated on the stack and could cause memory access issues.

Reply:
Yes, but only if the `Vector` is expected to be deleted during the lifetime of a `_CullConvexData`, which it isn't, because it only exists for the duration of the subsequent call to `_cull_convex()`. However, it's safer because it prevents a `null` pointer (or any other arbitrary pointer) being passed. In this case, for example, it prevents a pointer to the first element of the vector being passed, which may not exist even if the Vector does. This is what caused #38457.

However, your question does highlight that for some reason I didn't do the same with planes, which may explain @reduz's "well, its slower" comment. I'll fix that.